### ISSUE TITLE
Organization API

### ISSUE THREAD

**Issue by to-sta** [activist-org/activist#627]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

We are starting with creating the API's that will be used by the frontend in production soon 😃. 

We are using [django-rest-framework](https://www.django-rest-framework.org/) in the backend to create our API's. ViewSets in particular the [ModelViewSet](https://www.django-rest-framework.org/api-guide/viewsets/#modelviewset) are used for creating the different HTTP methods. Another benefit of using the ModelViewSet is that we are getting some validation from the model itself and only need to extend that in the serializer if necessary.

Have look at the [scheme](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=5377-96159&mode=design&t=8xoYzpakJ6ebt8Ie-4) and the models.py file. 

Task is to implement all these methods for the OrganizationViewSet that represent the different HTTP methods:

```python
def list(self, request):
    pass

def create(self, request):
    pass

def retrieve(self, request, pk=None):
    pass

def update(self, request, pk=None):
    pass

def partial_update(self, request, pk=None):
    pass

def destroy(self, request, pk=None):
    pass
```

The API generally is a combiniation of:

- views.py
- serializer.py
- models.py

Also feel free to suggest changes to the model, in case it is necessary for the logic. They are not set in stone 😃. In terms of testing, you can use Swagger UI for manuall testing (backend endpoint: `/v1/schema/swagger-ui/`).

>[!NOTE]
> Please use an appropiate http [status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) for your responses.

### Contribution

I am happy to help with this issue.

**Comment by andrewtavis** [comment]
I'll take this one, @to-sta 😊 We can also make ones for organizations and FAQs as well to round out the main things from the start, and then maybe events. As you see fit :) :)

**Comment by to-sta** [comment]
hmm I was thinking about starting with the main tables:

* resources
* organizations
* event
* topic
* task

What table/model represents the for example FAQ?

**Comment by bharath637462** [comment]
@to-sta can i create api for 
resources
organizations
event
topic
task
below tables?

**Comment by Jackblanket847** [comment]
def list(self, request):
    pass

def create(self, request):
    pass

def retrieve(self, request, pk=None):
    pass

def update(self, request, pk=None):
    pass

def partial_update(self, request, pk=None):
    pass

def destroy(self, request, pk=None):
    pass

**Comment by andrewtavis** [comment]
Hey @josueemartinezz 👋 I think that @to-sta is mostly finalizing this at this point. We'll be doing a sync to finalize this this weekend, but there should be more backend issues after that! If you find another, please write in and we'll do our best to assign you :)

**Comment by josueemartinezz** [comment]
> Hey @josueemartinezz 👋 I think that @to-sta is mostly finalizing this at this point. We'll be doing a sync to finalize this this weekend, but there should be more backend issues after that! If you find another, please write in and we'll do our best to assign you :)

Ah, I see, I'll keep an eye out for to-sta's pull request. Thanks for your response! Looking forward to contribute :)

**Comment by andrewtavis** [comment]
Looking forward to your contribution! :)

**Comment by andrewtavis** [comment]
Closed via #629. Thank you, @to-sta! 🙏🙏
